griptape/drivers/vector/qdrant_vector_store_driver.py

```diff
@@ -0,0 +1,227 @@
+            if (
+                deletion_response.status
+                == import_optional_dependency("qdrant_client.http.models").UpdateStatus.COMPLETED
+            ):
+                print(f"ID {vector_id} is successfully deleted")
+            else:
+                print(f"Failed to delete ID {vector_id}. Status: {deletion_response.status}")
+        except Exception as e:
+            print(f"An error occurred while trying to delete ID {vector_id}: {e}")
+
+    def query(
+        self,
+        query: str,
+        count: Optional[int] = None,
+        namespace: Optional[str] = None,
+        include_vectors: bool = False,
+        with_payload: bool = True,
+        with_vectors: bool = True,
+        **kwargs,
+    ) -> list[BaseVectorStoreDriver.QueryResult]:
+        """
+        Query the Qdrant collection based on a query vector.
+
+        Parameters:
+            query (str): Query string.
+            count (Optional[int]): Optional number of results to return.
+            namespace (Optional[str]): Optional namespace of the vectors.
+            include_vectors (bool): Whether to include vectors in the results.
+
+        Returns:
+            list[BaseVectorStoreDriver.QueryResult]: List of QueryResult objects.
+        """
+        query_vector = self.embedding_driver.embed_string(query)
+
+        # Create a search request
+        results = self.client.search(
+            collection_name=self.collection_name,
+            query_vector=query_vector,
+            limit=count,
+            with_payload=with_payload,
+            with_vectors=with_vectors,
+        )
+
+        # Convert results to QueryResult objects
+        query_results = [
+            BaseVectorStoreDriver.QueryResult(
+                id=result.id,
+                vector=result.vector if include_vectors else [],
+                score=result.score,
+                meta={k: v for k, v in result.payload.items() if k not in ["_score", "_tensor_facets"]},
+            )
+            for result in results
+        ]
+        return query_results
+
+    def upsert_vector(
+        self,
+        vector: list[float],
+        vector_id: Optional[str] = None,
+        namespace: Optional[str] = None,
+        meta: Optional[dict] = None,
+        content: Optional[str] = None,
+        **kwargs,
+    ) -> str:
+        """
+        Upsert vectors into the Qdrant collection.
+
+        Parameters:
+            vector (list[float]): The vector to be upserted.
+            vector_id (Optional[str]): Optional vector ID.
+            namespace (Optional[str]): Optional namespace for the vector.
+            meta (Optional[dict]): Optional dictionary containing metadata.
+            content (Optional[str]): The text content to be included in the payload.
+
+        Returns:
+            str: The ID of the upserted vector.
+        """
+
+        if vector_id is None:
+            vector_id = vector_id if vector_id else utils.str_to_hash(str(vector))
```

Comment:
Qdrant only accepts UUID and +ve integers as ID.

https://qdrant.tech/documentation/concepts/points/#point-ids

Reply:
Earlier had used the same, removed based on comment. Prev commit: https://github.com/griptape-ai/griptape/pull/812/commits/b2582a01a681296d934e2eae194cc1d16552885b

Re-added